File: 02_Object_Oriented_Programming/02_Class_Relationships/06_Realization/javascript/upiPayment.js
/**
 * Realization — UpiPayment in JavaScript.
 */

class UpiPayment {
  #upiId;

  constructor(upiId) {
    this.#upiId = upiId;
  }

  get paymentMethod() {
    return "UPI";
  }

  pay(amount) {
    console.log(`Paid $${amount} via UPI (${this.#upiId}).`);
  }
}

module.exports = UpiPayment;

File: 02_Object_Oriented_Programming/02_Class_Relationships/06_Realization/python/upi_payment.py
"""
Realization — UpiPayment implements Payment.
"""

from payment import Payment


class UpiPayment(Payment):
    """Pays using UPI."""

    def __init__(self, upi_id: str) -> None:
        self._upi_id = upi_id

    @property
    def payment_method(self) -> str:
        return "UPI"

    def pay(self, amount: float) -> None:
        print(f"Paid ${amount} via UPI ({self._upi_id}).")
